src/dev/Web/EastSeat.ResourceIdea.Web/Components/Pages/WorkItems/AddWorkItem.razor.cs

```diff
@@ -0,0 +1,296 @@
+                }
+            }
+            else
+            {
+                // Load all engagements for the tenant
+                var query = new GetAllEngagementsQuery(1, 100);
+                query.TenantId = ResourceIdeaRequestContext.Tenant;
+
+                var response = await Mediator.Send(query);
+                if (response.IsSuccess && response.Content.HasValue)
+                {
+                    Engagements = response.Content.Value;
+                }
+                else
+                {
+                    HasError = true;
+                    ErrorMessage = "Failed to load engagements. Please try again later.";
+                }
+            }
+        }
+        catch (InvalidOperationException ex)
+        {
+            HasError = true;
+            ErrorMessage = "An error occurred while loading engagements. Please try again later.";
+            Console.Error.WriteLine($"InvalidOperationException in LoadEngagements: {ex}");
+        }
+        catch (HttpRequestException ex)
+        {
+            HasError = true;
+            ErrorMessage = "A network error occurred while loading engagements. Please check your connection and try again.";
+            Console.Error.WriteLine($"HttpRequestException in LoadEngagements: {ex}");
+        }
+        catch (TimeoutException ex)
+        {
+            HasError = true;
+            ErrorMessage = "The request timed out while loading engagements. Please try again later.";
+            Console.Error.WriteLine($"TimeoutException in LoadEngagements: {ex}");
+        }
+        catch (Exception ex)
+        {
+            Console.Error.WriteLine($"Unexpected error in LoadEngagements: {ex}");
+            throw; // Rethrow unexpected exceptions to be handled at a higher level
+        }
+    }
+
+    private async Task LoadEmployees()
+    {
+        try
+        {
+            TenantEmployeesQuery query = new()
+            {
+                TenantId = ResourceIdeaRequestContext.Tenant,
+                PageNumber = 1,
+                PageSize = 100, // Get a larger page size to include all employees
+            };
+
+            var response = await Mediator.Send(query);
+            if (response.IsSuccess && response.Content.HasValue)
+            {
+                Employees = response.Content.Value;
+            }
+            else
+            {
+                // Note: It's OK if employees loading fails, as assignment is optional
+                Console.WriteLine("Failed to load employees, but continuing as assignment is optional.");
+            }
+        }
+        catch (InvalidOperationException ex)
+        {
+            Console.Error.WriteLine($"InvalidOperationException in LoadEmployees: {ex}");
+        }
+        catch (HttpRequestException ex)
+        {
+            Console.Error.WriteLine($"Network error in LoadEmployees: {ex}");
```

Comment:
I am going to setup logging using Azure Application Insights, don't use `Console.Error.WriteLine`.